sql/core/src/main/scala/org/apache/spark/sql/execution/GenerateExec.scala

```diff
@@ -57,20 +62,19 @@ private[execution] sealed case class LazyIterator(func: () => TraversableOnce[In
  */
 case class GenerateExec(
     generator: Generator,
-    join: Boolean,
+    unrequiredChildIndex: Seq[Int],
```

Comment:
The physical plan can just take `requiredChildOutput`, and in the planner we can just do
```
case g @ logical.Generate(...) => GenerateExec(..., g.requiredChildOutput)
```